File: Kahla.SDK/Models/ApiViewModels/ContactInfo.cs
using Newtonsoft.Json;

namespace Kahla.SDK.Models.ApiViewModels
{
    public class ContactInfo
    {
        public string DisplayName { get; set; }
        public string DisplayImagePath { get; set; }
        public Message LatestMessage { get; set; }
        public int UnReadAmount { get; set; }
        public int ConversationId { get; set; }
        public string Discriminator { get; set; }
        public string UserId { get; set; }
        public string AesKey { get; set; }
        public bool Muted { get; set; }
        public bool SomeoneAtMe { get; set; }
        public bool Online { get; set; }
        [JsonIgnore]
        public bool EnableInvisiable { get; set; }
        [JsonIgnore]
        public KahlaUser Sender { get; set; }
    }
}

File: src/app/Models/ContactInfo.ts
﻿import { Message } from './Message';

export class ContactInfo {
    public displayName: string;
    public displayImagePath: string;
    public latestMessage: Message;
    public unReadAmount: number;
    public conversationId: number;
    public discriminator: 'GroupConversation' | 'PrivateConversation';
    public userId: string;
    public avatarURL: string;
    public muted: boolean;
    public someoneAtMe: boolean;
    public online?: boolean;
}
